src/main/scala/org/constellation/invertedmap/Inverted.scala

```diff
@@ -0,0 +1,40 @@
+package org.constellation.invertedmap
+
+object Inverted {
+  def isEmpty(m: Map[_, _]): Boolean = m.isEmpty
+
+  def contains[A](m: Map[_, Set[A]], x: A): Boolean =
+    m.exists { case (_, v) => v.contains(x) }
+
+  def get[A, B](m: Map[B, Set[A]], x: A): Option[B] =
+    m.find { case (_, v) => v.contains(x) }.map { case (k, _) => k }
+
+  def update[A, B, B1 >: B](m: Map[B1, Set[A]], k: A, v: B): Map[B1, Set[A]] = {
+    val keys = m.getOrElse(v, Set.empty)
+    m.updated(v, keys ++ Set(k))
+  }
+
+  def delete[A, B](m: Map[B, Set[A]], x: A): Map[B, Set[A]] =
+    m.find { case (_, v) => v.contains(x) }.map {
```

Comment:
Doesn't find return the first optional value?
Wouldn't it work nondeterministically removing value from the first key which Set contains the value?
Shouldn't it remove the value from all sets, or get a key parameter to remove from a Set under a given key?

Reply:
there is an assumption (and assertion) that `A` can be present in a single Set only

Reply:
```
InvertedMap(("node1", allReady), ("node2", allReady), ("node3", oneOffline), ("node3", allReady))
```
as well as
```
m + (("node3", allReady))
```
throws an error

Reply:
I'm not sure if it should throw or just not change the map/take the first value?
I'm thinking about if we should be failing the flow/handling the exception.
If yes maybe the return type should indicate that it may fail?

Reply:
We have `Map[A, B]` so `A` is a key - that's the assumption (map has unique keys).
In the underlying map it's in value position as `Set[A]` so we have to assert for that or addition of the same key should remove it from the previous place but it would be an implicit behavior and imo we should not do that.

Anyway, it's commonly done that way. See: insert into a `TreeMap`.
```
  def insert [B1 >: B](key: A, value: B1): TreeMap[A, B1] = {
    assert(!RB.contains(tree, key))
    new TreeMap(RB.update(tree, key, value, overwrite = true))
  }
```